consensus/istanbul/core/handler.go

```diff
@@ -219,3 +224,15 @@ func (c *core) handleTimeoutMsg(desiredView *istanbul.View) error {
 	nextRound := new(big.Int).Add(desiredView.Round, common.Big1)
 	return c.waitForDesiredRound(nextRound)
 }
+
+func (c *core) handleResendRoundChangeEvent(desiredView *istanbul.View) error {
+	logger := c.newLogger("func", "handleResendRoundChangeEvent", "set_at_seq", desiredView.Sequence, "set_at_desiredRound", desiredView.Round)
+
+	if c.current.Sequence().Cmp(desiredView.Sequence) != 0 || c.current.DesiredRound().Cmp(desiredView.Round) != 0 {
```

Comment:
If `currentRound == desiredRound`, doesn't that mean we have advanced to the desired round and shouldn't send a round change message? I would assume we would send a round change here iff `currentRound < desiredRound`

Reply:
This is specifically to catch race condition of event being executed but view having changed in the mean time. I will add a comment.

Reply:
Added a comment.  This check is only to prevent races since other messages could  be processed between enqueueing this timeout and processing the event. Equality with both means there has not been a race with updating the sequence or advancing the actual or desired view elsewhere - in those cases this event should be suppressed and we shouldn't send a RoundChange as a result.